File: assignment.py
class StatisticsCalculator:
    def __init__(self, lst):
        self.lst = lst

    def calculate_mean(self):
        return sum(self.lst) / len(self.lst)

    def calculate_median(self):
        sorted_lst = sorted(self.lst)
        mid = len(self.lst) // 2
        if len(self.lst) % 2 == 0:
            return (sorted_lst[mid - 1] + sorted_lst[mid]) / 2.0
        else:
            return sorted_lst[mid]

    def calculate_mode(self):
        count = {}
        for num in self.lst:
            if num in count:
                count[num] += 1
            else:
                count[num] = 1
        max_count = max(count.values())
        mode = [num for num, freq in count.items() if freq == max_count]
        return mode[0]

# Main function
if __name__ == "__main__":
    lst = [1, 2, 3, 4, 5]
    calculator = StatisticsCalculator(lst)
    mean = calculator.calculate_mean()
    median = calculator.calculate_median()
    mode = calculator.calculate_mode()
    print(f"Mean: {mean:.2f}")
    print(f"Median: {median:.2f}")
    print(f"Mode: {mode}")

File: assignment.c
#include <stdio.h>
#include <stdlib.h>

// Function to calculate mean
double calculate_mean(int arr[], int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum += arr[i];
    }
    return (double)sum / size;
}

// Function to calculate median
double calculate_median(int arr[], int size) {
    // Sort the array
    for (int i = 0; i < size - 1; i++) {
        for (int j = i + 1; j < size; j++) {
            if (arr[i] > arr[j]) {
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }

    // Calculate median
    if (size % 2 == 0) {
        return (arr[size / 2 - 1] + arr[size / 2]) / 2.0;
    } else {
        return arr[size / 2];
    }
}

// Function to calculate mode
int calculate_mode(int arr[], int size) {
    int max_count = 0;
    int mode = arr[0];

    for (int i = 0; i < size; i++) {
        int count = 0;
        for (int j = 0; j < size; j++) {
            if (arr[i] == arr[j]) {
                count++;
            }
        }
        if (count > max_count) {
            max_count = count;
            mode = arr[i];
        }
    }

    return mode;
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int size = sizeof(arr) / sizeof(arr[0]);

    double mean = calculate_mean(arr, size);
    double median = calculate_median(arr, size);
    int mode = calculate_mode(arr, size);

    printf("Mean: %.2f\n", mean);
    printf("Median: %.2f\n", median);
    printf("Mode: %d\n", mode);

    return 0;
}